### PR TITLE
Fix #6 - Support for webviews

### LINKED ISSUE #6: Doesn't work inside <webview>
The context menu opens correctly when a right-clicked occurs inside a `webview`, but the inspector opens up to show the parent context, not the `webview`'s context:

image

It would be more useful if the `webview`'s inspector tools were automatically opened.

### PR DESCRIPTION
**Pull request by teameh**
See https://github.com/parro-it/debug-menu/issues/6

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -4,13 +4,13 @@ const electron = require('electron');
...
+function inpectMenuTemplate(elm, pos) {
   return {
     label: 'Inspect element',
     click: () => {
-      electron.remote
-        .getCurrentWindow()
-        .inspectElement(pos.x, pos.y);
+        const win = elm.tagName.toLowerCase() === 'webview' ?
+            elm : electron.remote.getCurrentWindow()
+        win.inspectElement(pos.x, pos.y);
```
Comment (teameh) [inline index.js]: or

```
(elm.tagName.toLowerCase() === 'webview' ? elm : electron.remote.getCurrentWindow())
    .inspectElement(pos.x, pos.y);
```

or

```
(
    elm.tagName.toLowerCase() === 'webview' ? 
        elm : 
        electron.remote.getCurrentWindow()
).inspectElement(pos.x, pos.y);
```

Comment (parro-it) [reply index.js]: I personally prefer this format:
```js
(
    elm.tagName.toLowerCase() === 'webview' ? 
        elm : 
        electron.remote.getCurrentWindow()
).inspectElement(pos.x, pos.y);
```

But it's better if you switch the arguments order, first the `pos`, then `elm`, so we keep compatibility with previous version. You will have then to check the elm argument: if it is set, use it, otherwiese if undefined, use the current window.

Comment (parro-it) [reply index.js]: you can avoid the tagname check, we can give for grant that the user is passing an element that has an `inspectElement` method.

**Comment by parro-it** [comment]
Thank you for the PR!

**Comment by parro-it** [comment]
Please see ethe comment on the arguments order, and it would be beautiful if you could add the option to the install method too

File: index.js (hunk 1 above)
Comment (teameh) [reply index.js]: Ah you mean just this ?

```
    elm..inspectElement(pos.x, pos.y);
```

We can skip the `electron.remote.getCurrentWindow()` all together? Okay :) will fix it tomorrow.

**Comment by teameh** [comment]
> Please see ethe comment on the arguments order

Will fix the order tomorrow. 

What do you mean with: 

>  could you add the option to the install method too ?

**Comment by parro-it** [comment]
> What do you mean with:
could you add the option to the install method too ?

There is an install `method` that allow to use `debug-menu` independently from `electron-contextmenu-middleware`. IT would be useful to add the same option there


> Ah you mean just this ?
    elm..inspectElement(pos.x, pos.y);
We can skip the electron.remote.getCurrentWindow() all together? Okay :) will fix it tomorrow.

No, I mean 

```js
(elm || electron.remote.getCurrentWindow()).inspectElement(pos.x, pos.y);
```

That is, we use `elm` if the user provide it, otherwise we default to current window

**Comment by parro-it** [comment]
Also, I had done some investigation using `webContents.getFocusedWebContents()`.
It successfully return the webContents of webviews when they are focused, without requiring the user to provide additional options.

But the `pos` argument contains coordinates referred to the host document, so `inspectElement` is called on wrong element. However, I think this could be easily solved (maybe in `electron-contextmenu-middleware`).

Have a look at the `experiment-getFocusedWebContents` branch, what do you think about?

**Comment by teameh** [comment]
> Please see the comment on the arguments order

Fixed

> There is an install method that allow to use debug-menu independently from electron-contextmenu-middleware. IT would be useful to add the same option there

Done but can't test it right now.

**Comment by teameh** [comment]
> Have a look at the experiment-getFocusedWebContents branch, what do you think about?

Looks good 👍

**Comment by teameh** [comment]
@parro-it did you had any time to check this out?

**Comment by teameh** [comment]
@parro-it ? It would be great if you could merge and release this! :)

**Comment by parro-it** [comment]
Sorry for the long await @tiemevanveen.
I merged you commits, thank you so much for the PR.
I had to add a check to make sure that the actual element received has an inpectElement method.
If I omit this check, I get an exception when right clicking on normal element (check on file test.js).

**Comment by teameh** [comment]
Great! Thanks :)!

**Pull request merged**